worlds/oot_soh/location_access/dungeons/dodongos_cavern.py

```diff
@@ -4,219 +4,406 @@
+    ])
+    # Locations
+    add_locations(Regions.DODONGOS_CAVERN_FAR_BRIDGE, world, [
+        (Locations.DODONGOS_CAVERN_END_OF_BRIDGE_CHEST, lambda bundle: can_break_mud_walls(bundle))
+    ])
+    # Connections
+    connect_regions(Regions.DODONGOS_CAVERN_FAR_BRIDGE, world, [
+        (Regions.DODONGOS_CAVERN_BOMB_ROOM_UPPER, lambda bundle: True),
+        (Regions.DODONGOS_CAVERN_LOBBY, lambda bundle: True),
+    ])
+
+    ## Dodongos Cavern Boss Region
+    # Events
+    add_events(Regions.DODONGOS_CAVERN_BOSS_REGION, world, [
+        (EventLocations.DODONGOS_CAVERN_FAIRY_POT, Events.CAN_ACCESS_FAIRIES, lambda bundle: True),
+    ])
+    # Locations
+    add_locations(Regions.DODONGOS_CAVERN_BOSS_REGION, world, [
+        (Locations.DODONGOS_CAVERN_BEFORE_BOSS_GRASS, lambda bundle: can_cut_shrubs(bundle)),
+    ])
+    # Connections
+    connect_regions(Regions.DODONGOS_CAVERN_BOSS_REGION, world, [
+        (Regions.DODONGOS_CAVERN_LOBBY, lambda bundle: True),
+        (Regions.DODONGOS_CAVERN_BACK_ROOM, lambda bundle: can_break_mud_walls(bundle)),
+        (Regions.DODONGOS_CAVERN_BOSS_ENTRYWAY, lambda bundle: True),
+    ])
+
+    ## Dodongos Cavern Back Room
+    # Locations
+    add_locations(Regions.DODONGOS_CAVERN_BACK_ROOM, world, [
+        (Locations.DODONGOS_CAVERN_GS_BACK_ROOM, lambda bundle: can_attack(bundle)),
+        (Locations.DODONGOS_CAVERN_BACK_ROOM_POT1, lambda bundle: can_break_pots(bundle)),
+        (Locations.DODONGOS_CAVERN_BACK_ROOM_POT2, lambda bundle: can_break_pots(bundle)),
+        (Locations.DODONGOS_CAVERN_BACK_ROOM_POT3, lambda bundle: can_break_pots(bundle)),
+        [Locations.DODONGOS_CAVERN_BACK_ROOM_POT4, lambda bundle: can_break_pots(bundle)]
+    ])
+    # Connections
+    connect_regions(Regions.DODONGOS_CAVERN_BACK_ROOM, world, [
+        (Regions.DODONGOS_CAVERN_BOSS_REGION, lambda bundle: True),
+    ])
+
+    ## Dodongos Cavern Boss Entryway
+    # Connections
+    connect_regions(Regions.DODONGOS_CAVERN_BOSS_ENTRYWAY, world, [
+        (Regions.DODONGOS_CAVERN_BOSS_ROOM, lambda bundle: True),
+    ])
+
+    # Dodongos Cavern Boss Exit
+    # Connections
+    connect_regions(Regions.DODONGOS_CAVERN_BOSS_EXIT, world, [
+        (Regions.DODONGOS_CAVERN_BOSS_REGION, lambda bundle: True),
+    ])
+
+    ## Dodongos Cavern Boss Room
+    # Events
+    add_events(Regions.DODONGOS_CAVERN_BOSS_ROOM, world, [
+        (EventLocations.DODONGOS_CAVERN_BOSS_DEFEATED, Events.CLEARED_DODONGOS_CAVERN,
+         lambda bundle: has_explosives(bundle) or (
+                     (can_do_trick(Tricks.DC_HAMMER_FLOOR, bundle) and can_use(Items.MEGATON_HAMMER, bundle)) or (
+                         can_do_trick(Tricks.BLUE_FIRE_MUD_WALLS, bundle) and can_use(Items.BOTTLE_WITH_BLUE_FIRE,
+                                                                                      bundle))) and can_kill_enemy(
+             bundle, Enemies.KING_DODONGO)),
+    ])
```

Comment:
This isn't equal to the original. In the original they use a ternary operator to inline do an if else checking for the trick option RT_DC_HAMMER_FLOOR. This would be the closest python equivalent as far as I understand.
```suggestion
    add_events(Regions.DODONGOS_CAVERN_BOSS_ROOM, world, [
        (EventLocations.DODONGOS_CAVERN_BOSS_DEFEATED, Events.CLEARED_DODONGOS_CAVERN,
         lambda bundle: has_explosives(bundle) or (can_use(Items.MEGATON_HAMMER, bundle) or (can_do_trick(Tricks.BLUE_FIRE_MUD_WALLS, bundle) and blue_fire(bundle)) if can_do_trick(Tricks.DC_HAMMER_FLOOR, bundle) else can_do_trick(Tricks.BLUE_FIRE_MUD_WALLS, bundle) and can_use(Items.BOTTLE_WITH_BLUE_FIRE, bundle)) and can_kill_enemy(Enemies.KING_DODONGO, bundle))
    ])
```

Reply:
Please double check.